synapse/storage/database.py

```diff
@@ -796,6 +800,20 @@ async def runInteraction(
             The result of func
         """
 
+        async def _run_callbacks(callbacks: List[_CallbackListEntry]):
+            sync_callbacks: List[_CallbackListEntry] = []
+
+            for cb, args, kwargs in callbacks:
+                if inspect.iscoroutinefunction(cb):
```

Comment:
is there a good reason to limit this to coroutine functions, rather than other awaitables? I think `inspect.isawaitable(cb)` would be better?